pandas/_libs/tslibs/np_datetime.pyx

```diff
@@ -47,6 +27,34 @@ cdef extern from "../src/datetime/np_datetime.h":
     pandas_datetimestruct _NS_MIN_DTS, _NS_MAX_DTS
 
 # ----------------------------------------------------------------------
+# numpy object inspection
+
+cdef inline npy_datetime get_datetime64_value(object obj) nogil:
+    """
```

Comment:
Can you make the docstrings more descriptive?  Something like `"returns int64 value underlying numpy datetime scalar Python object"`

Reply:
Good idea, will change.